src/manager.rs

```diff
@@ -407,6 +354,128 @@ impl ManagerService for Arc<Manager> {
     }
 }
 
+fn compute_quorum_results(
+    replica_id: &str,
+    rank: i64,
+    quorum: &Quorum,
+) -> Result<ManagerQuorumResponse, Status> {
+    let mut participants = quorum.participants.clone();
+    participants.sort_by(|a, b| a.replica_id.cmp(&b.replica_id));
+
+    // Compute the rank of the replica in the returned quorum.
+    let replica_rank = participants
+        .iter()
+        .enumerate()
+        .find_map(|(i, p)| {
+            if p.replica_id == replica_id {
+                Some(i)
+            } else {
+                None
+            }
+        })
+        .ok_or_else(|| {
+            Status::not_found(format!(
+                "replica {} not participating in returned quorum",
+                replica_id
+            ))
+        })?;
+
+    let step = participants[replica_rank].step;
+
+    // Compute the details for workers at max step.
+    let max_step = participants.iter().map(|p| p.step).max().unwrap();
+    let max_participants: Vec<&QuorumMember> =
```

Comment:
just checking, once we reach into compute_quorum_results will there always be a nonzero number of ranks at max_step in the quorum?

Reply:
yes, there's always at least 1 member at max_step since it's defined as the maximum step of participating members and we'll always have at least 1 member